openshift/WFLY-15421_OIDC_for_WildFly_s2i_v2.adoc

```diff
@@ -0,0 +1,297 @@
+
+|OIDC_PROVIDER_NAME
+|NONE. When SSO_* env var are used, “keycloak” name is internally set.
+|To identify the provider to use.
+|Y
+|
+
+|OIDC_PROVIDER_URL
+|`$SSO_URL/realms/$SSO_REALM`
+|The url of the provider
+|Y
+|
+
+|OIDC_USER_NAME
+|SSO_USERNAME
+|User name used to get a token, used for dynamic client registration.
+|Y
+|
+
+|OIDC_USER_PASSWORD
+|SSO_PASSWORD
+|User password used to get a token, used for dynamic client registration.
+|Y
+|
+
+|OIDC_SECURE_DEPLOYMENT_SECRET
+|SSO_SECRET
+|Secure deployment secret known in the subsystem secure-deployment and the auth server client.
+|N
+|
+
+|OIDC_SECURE_DEPLOYMENT_PRINCIPAL_ATTRIBUTE
+|SSO_PRINCIPAL_ATTRIBUTE
+|Configure the principal value of the principal name.
+|N
+|Default to sub (token id) for keycloak, typical value is preferred_username 
+
+|OIDC_SECURE_DEPLOYMENT_ENABLE_CORS
+|SSO_ENABLE_CORS
+|Enable CORS
+|N
+|False by default 
+
+|OIDC_SECURE_DEPLOYMENT_BEARER_ONLY
+|SSO_BEARER_ONLY
+|Deployment that accepts Bearer token only, no logging.
+|N
+|False by default
+
+|OIDC_PROVIDER_SSL_REQUIRED
+|NONE, new env variable (see CLOUD-3529 )
+|Is https enabled in the realm. By default external (private and local address no https).
+|N
+|external
+
+|OIDC_PROVIDER_TRUSTSTORE
+|SSO_TRUSTSTORE
+|Name of the realm trustore file. If not set, disable-trust-manager=true
+|N
+|
+
+|OIDC_PROVIDER_TRUSTSTORE_DIR
+|SSO_TRUSTSTORE_DIR
+|Directory to find the realm truststore. If not set, disable-trust-manager=true
+|N
+|
+
+|OIDC_PROVIDER_TRUSTSTORE_PASSWORD
+|SSO_TRUSTSTORE_PASSWORD
+|Realm Truststore password. If not set, disable-trust-manager=true
+|N
+|
+
+|OIDC_PROVIDER_TRUSTSTORE_CERTIFICATE_ALIAS
+|SSO_TRUSTSTORE_CERTIFICATE_ALIAS
+|Realm trustore alias, required to interact with auth server to register client
+|N
+|
+
+|OIDC_DISABLE_SSL_CERTIFICATE_VALIDATION
+|SSO_DISABLE_SSL_CERTIFICATE_VALIDATION
+|Disable certificate validation when interacting with auth server to register client.
+|N
+|
+
+|OIDC_HOSTNAME_HTTP
+|HOSTNAME_HTTP
+|Hostname used for unsecure routes 
+|N
+|Routes are discovered
+
+|OIDC_HOSTNAME_HTTPS
+|HOSTNAME_HTTPS
+|Hostname used for secured routes
+|N
+|Secured routes are discovered
+
+|NONE
+|SSO_PUBLIC_KEY
+|Realm public key. This option is no more used, public key being automatically retrieved by the OIDC subsystem.
+|N
+|If set a warning is displayed to advertise that this option is being ignored.
+
+|===
+
+
+===== Dynamic registration of clients
+
+This registration is enabled only for the `keycloak` provider. Other providers would need the client to be already 
+registered inside the authorization server.
+
+===== Routes discovery
+
+* Route discovery (if no OIDC_HOSTNAME_HTTP(S) is set) is bound to openshift. 
+* This feature can’t be used on other cloud system.
+* The route discovery  implies that complex JSON structure be parsed.  `jq` (small runtime to parse json returned by openshift service)  is expected to be 
+installed in the WildFly s2i v2 image.
```

Comment:
In which image? the builder or the runtime image?

Reply:
Both, fixed.